canister/src/block_header_store.rs

```diff
@@ -71,6 +71,14 @@ impl BlockHeaderStore {
                 .expect("block header must exist")
         })
     }
+
+    /// Returns block headers in the range [start_hegiht, end_height].
+    pub fn get_block_headers_in_range(&self, start_height: u32, end_height: u32) -> Vec<Vec<u8>> {
```

Comment:
All the methods in this component are dealing with block headers as either `BlockHeaderBlob` or `BlockHeader`. It's confusing/inconsistent to have this method now return a block header as `Vec<u8>`. That's why my recommendation is to have the following signature:

```
    pub fn get_block_headers_in_range(
        &self,
        heights: std::ops::RangeInclusive<Height>,
    ) -> impl Iterator<Item = BlockHeaderBlob> + '_ {
```

This function returns an iterator of `BlockHeaderBlob`, when then in `get_block_headers` you can `map` into a `Vec<u8>`.

Reply:
Done.